File: HaywireGM.Contracts/DbEntities/Npc.cs
namespace HaywireGM.Contracts.DbEntities;

public class Npc
{
    public int npc_id { get; set; }
    public int account_id { get; set; }
    public required int campaign_id { get; set; }
    // Core fields
    public required string name { get; set; }
    public string? description { get; set; }
    // Flattened properties (moved from stats object)
    public string? lineage { get; set; }
    public string? @class { get; set; }
    public string? faction { get; set; }
    public string? notes { get; set; }
}

File: HaywireGM.React/src/types/npc.ts
import type { EntityType } from './entity';

export interface NPC {
  id: number;
  name: string;
  lineage: string;
  class: string;
  description: string;
  campaignId?: number;
  faction?: string;
  notes?: string;
  accountId?: number;
}

export interface Relationship {
  id: number;
  npcId1: number;
  npcId2: number;
  entityType1: EntityType;
  entityType2: EntityType;
  type: RelationshipType;
  description?: string;
  attitudeScore: number;
  campaignId?: number;
  isDerived?: boolean;
}

export type RelationshipType =
  | 'acquaintance'
  | 'ally'
  | 'friend'
  | 'contact/informant'
  | 'employer'
  | 'enemy'
  | 'family'
  | 'lover'
  | 'member'
  | 'mentor'
  | 'patron'
  | 'rival'
  | 'stranger'
  | 'vassal/follower'
  | 'neutral';

// Auth context types
export interface User {
  cognitoSub: string;
  email: string;
  accountId?: number;
}

export interface AuthState {
  isAuthenticated: boolean;
  user: User | null;
  loading: boolean;
  isLoggingIn: boolean;
}
